validator/sawtooth_validator/state/batch_tracker.py

```diff
@@ -61,12 +62,24 @@ def chain_update(self, block, receipts):
         """Removes batches from the pending cache if found in the block store,
         and notifies any observers.
         """
+        to_be_notified = []
         with self._lock:
-            for batch_id in self._pending.copy():
-                if self._block_store.has_batch(batch_id):
-                    self._pending.remove(batch_id)
-                    self._update_observers(batch_id,
-                                           ClientBatchStatus.COMMITTED)
+            committed_txn_ids = [receipt.transaction_id
+                                 for receipt in receipts]
+
+            for txn_id in committed_txn_ids:
```

Comment:
Can you add a comment clarifying that this works because batches are committed atomically?